atc/policy/opa/opa.go

```diff
@@ -86,10 +86,8 @@ func (c opa) Check(input policy.PolicyCheckInput) (policy.PolicyCheckOutput, err
 		return policy.FailedPolicyCheck(), fmt.Errorf("opa returned bad response: %s", err.Error())
 	}
 
-	// If no result returned, meaning that the requested policy decision is
-	// undefined OPA, then consider as pass.
-	if result.Result == nil {
-		return policy.PassedPolicyCheck(), nil
+	if result.Result == nil || result.Result.Allowed == nil {
+		return policy.FailedPolicyCheck(), fmt.Errorf("opa returned invalid response: %s", resp.Body)
```

Comment:
I got a really intelligible response back by creating a rule that looked like this:

```rego
default decision = {"reasons": ["hey there"]}
``` 
In my build page I got this, which looks like the struct of an `io.ReadCloser`, which is what `Response.Body` is.
```
opa returned invalid response: &{%!s(func()=0xf65060) %!s(*http.bodyEOFSignal=&{0xc000a3d140 {0 0} false 0xc000118100 <nil> 0x15ff940}) %!s(func() bool=0x15f82a0)}
```

I changed it to return the `body` var on line 78
```suggestion
		return policy.FailedPolicyCheck(), fmt.Errorf("opa returned invalid response: %s", body)
```
Which gives a more human readable message:
```
opa returned invalid response: {"result":{"reasons":["hey there"]}}
```